docs/api.rst

```diff
@@ -5166,6 +5166,14 @@ AllowedMentions
 .. autoclass:: AllowedMentions
     :members:
 
+MessageSnapshot
+~~~~~~~~~~~~~~~~~
+
+.. attributetable:: MessageSnapshot
+
+.. autoclass:: MessageSnapshot
+    :members:
+
```

Comment:
Since this is an API received object it needs to be in the model section with the constructor hidden.